enginetest/queries/json_scripts.go

```diff
@@ -333,6 +333,35 @@ var JsonScripts = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "JSON -> and ->> operator support",
+		SetUpScript: []string{
+			"create table t (pk int primary key, col1 JSON);",
+			`insert into t values (1, JSON_OBJECT('key1', 1, 'key2', '"abc"'));`,
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    `select col1->'$.key1' from t;`,
+				Expected: []sql.Row{{types.JSONDocument{Val: 1}}},
+			},
+			{
+				Query:    `select col1->>'$.key2' from t;`,
+				Expected: []sql.Row{{"abc"}},
+			},
+			{
+				Query:    `select * from t where col1->'$.key1' = 1;`,
+				Expected: []sql.Row{{1, types.JSONDocument{Val: map[string]interface{}{"key1": 1, "key2": "\"abc\""}}}},
+			},
+			{
+				Query:    `select * from t where col1->>'$.key2' = 'abc';`,
+				Expected: []sql.Row{{1, types.JSONDocument{Val: map[string]interface{}{"key1": 1, "key2": "\"abc\""}}}},
+			},
+			{
+				Query:    `select * from t where col1->>'$.key2' = 'def';`,
+				Expected: []sql.Row{},
+			},
```

Comment:
Might be worth adding these test cases

Check if subqueries/aliasing works, and if operators can be nested
```sql
CREATE TABLE t (i INT PRIMARY KEY, j JSON);
INSERT INTO t VALUES (0, '{"a": "123", "outer": {"inner": 456}}');
SELECT k->"$.inner" from (SELECT j->"$.outer" AS k FROM t) sq;
+--------------+
| k->"$.inner" |
+--------------+
| 456          |
+--------------+
```

Check if array operations work
```sql
CREATE TABLE tj10 (a JSON, b INT);
INSERT INTO tj10 VALUES ('[3,10,5,"x",44]', 33), ('[3,10,5,17,[22,"y",66]]', 0);
SELECT a->"$[3]", a->"$[4][1]" FROM tj10;
+-----------+--------------+
| a->"$[3]" | a->"$[4][1]" |
+-----------+--------------+
| "x"       | NULL         |
| 17        | "y"          |
+-----------+--------------+

SELECT a->>"$[3]", a->>"$[4][1]" FROM tj10;
+------------+---------------+
| a->>"$[3]" | a->>"$[4][1]" |
+------------+---------------+
| x          | NULL          |
| 17         | y             |
+------------+---------------+
```

Reply:
Nice! 😄 Test coverage for JSON arrays was the missing piece I was hoping you'd find. Good job! 😄 

I like the subquery/aliasing suggestion, too. I'll add that in next.